java-checks/src/main/java/org/sonar/java/checks/UnnecessaryEscapeSequencesInTextBlockCheck.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.java.collections.SetUtils;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S5665")
+public class UnnecessaryEscapeSequencesInTextBlockCheck extends IssuableSubscriptionVisitor {
+  private static final String MESSAGE = "Remove this unnecessary escape sequence.";
+  private static final String TRIPLE_QUOTE_MESSAGE = "Use '\\\"\"\"' to escape \"\"\".";
+  private static final Set<String> ESCAPED = SetUtils.immutableSetOf("\\n", "\\t", "\\'", "\\\"");
```

Comment:
I wouldn't include `\t` in the list of escapes we complain about. I feel that using `\t` is a much better idea than putting literal tab characters into your string - whether it's inside a normal string literal or a text block.

Reply:
This does make sense, thanks